### PR TITLE
[DS][25/n] Narrow extra_state type

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title -- we only need a subset of the full PackableValue type, and can always widen this later

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21613?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#21613** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21613?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21612** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21612?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21573** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21573?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21546** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21546?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21545** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21545?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21541** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21541?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21540** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21540?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21539** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21539?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21538** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21538?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21537** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21537?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21536** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21536?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21535** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21535?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21521** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21521?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21520** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21520?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21511** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21511?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#21512](https://github.com/dagster-io/dagster/pull/21512) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21512?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#21510** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21510?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21508** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21508?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21507** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21507?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21505** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21505?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21504** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21504?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21503** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21503?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21502** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21502?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21501** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21501?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21500** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21500?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21499** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21499?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/declarative_scheduling/serialized_objects.py [hunk 1]
```diff
@@ -31,6 +30,10 @@
 
 T = TypeVar("T")
 
+# state objects that store arbitrary data between ticks
+# if a need for new types arises, they can be added here, as long as they are serializable
+SerializableState = Union[None, AssetSubset, Sequence[ValidAssetSubset]]
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/definitions/declarative_scheduling/serialized_objects.py:35]: I don't think we should assign a name to this union. It's too general of name given what it is.

Can we just canonicalize to 'Optional[Sequence[ValidAssetSubset]]`?

**Review by schrockn — CHANGES_REQUESTED** [review]
back to your q. the type of this raises additional questions

**Comment by OwenKephart** [comment]
Fine with not giving it a name, but I don't want to mess with disallowing a singular AssetSubset at the moment (as we have plenty of already-stored information that is just a single AssetSubset)

**Comment by schrockn** [comment]
Makes complete sense to not change then given that we are dealing with already persisted data  👍🏻

**Review by schrockn — APPROVED** [review]

**Comment by OwenKephart** [comment]
### Merge activity

* **May 3, 5:31 PM EDT**: @OwenKephart started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21613).

**Pull request merged**